internal/conf/configuration.go

```diff
@@ -102,14 +102,53 @@ type JWTConfiguration struct {
 	KeyID            string   `json:"key_id" split_words:"true"`
 }
 
+type MFAFactorTypeConfiguration struct {
+	EnrollEnabled bool `split_words:"true" default:"true"`
+	VerifyEnabled bool `split_words:"true" default:"true"`
+}
+
+type WebauthnConfiguration struct {
+	MFAFactorTypeConfiguration
+	// TODO: decide whether to remove this later
+	Enabled bool `default:"false"`
+	// TODO: Come up with more sensible defaults for these
+	RPDisplayName string   `json:"rp_display_name" split_words:"true" default:"Go Webauthn"`
+	RPID          string   `json:"rp_id"`
+	RPOrigins     []string `json:"rp_origins" split_words:"true"`
+	// TODO: Parse and set this from config
+	Webauthn *webauthn.WebAuthn `json:"overall_configuration" split_words:"true"`
+}
+
+func (w *WebauthnConfiguration) PopulateWebauthnConfiguration() error {
+	// TODO: Decide whether to remove since there's already vlaidation in .New at library level
+	if w.RPDisplayName == "" || w.RPID == "" || len(w.RPOrigins) == 0 {
+		return fmt.Errorf("display name, ID, and origins need to be populated")
+	}
+	wconfig := &webauthn.Config{
+		RPDisplayName: w.RPDisplayName,
+		RPID:          w.RPID,
+		RPOrigins:     w.RPOrigins,
+	}
```

Comment:
Right so these, can't they be specified in the `POST /enroll` call?

Reply:
This is the backend server's security policy about what origins it can interact with. 

A concern here is that if we expose it on the frontend a separate malicious client could call the `/enroll` endpoint and claim a fake origin (e.g.  Say our site is `supabase.com` , a client from `magicschoolbus.com` could pretend to have a origin not representative of the site it is served from

```
createClient({<url>, <apikey}, {
    mfa: {
        RPOrigins: 'supabase.com',
        ...
    }
}) 
```

(Above not representative of actual client bindings)

We can of course enforce checks so that this isn't possible but storing the config on backend eliminates this possibility.

Reply:
As discussed, let's move this to the FE. Decision Rationale is on Notion RFC